### PR TITLE
Test/login script

### PR DESCRIPTION
**Pull request by clmedders**
**_Delete all details that do not apply to this PR!_**

## Jira ticket (required)
[Insert Jira ticket link
](https://bixal-projects.atlassian.net/jira/software/c/projects/VOTE/boards/254?modal=detail&selectedIssue=VOTE-944&assignee=611272e0eb7aef00694b8a97)
## Description (optional)
This Pr will add log in scripts for different roles.  The login info is hosted in the `cyrpess.config.js` file and then the custom commands are in the `commands.js` file. The commands will log in before the test, will run the test and then will delete the user after as well. 

### Testing steps
1. pull down branch 
2. start local 
3. cd into `testing` 
4. run `npm run cy:open` 
5. go to `backend` folder and there will be 3 tests (site admin, content editor and content manager)
6. ensure the tests pass and allow you to log in, right now it is just getting an element on the page.

### CODE REVIEW

File: testing/cypress.config.js [hunk 1]
```diff
@@ -14,7 +14,22 @@ reporter: 'cypress-mochawesome-reporter',
...
+      "admin_role": "site_admin",
+      // info for content editor role
+      "content_editor": "editor_test",
+      "editor_password": "vote1234",
+      "editor_role": "content_editor",
+      // info for content mananger role 
+      "content_manager": "manager_test",
+      "manager_password": "vote123",
+      "manager_role": "content_manager",
+    },
```
Comment (rayestrada) [inline testing/cypress.config.js:31]: Update the object to look like this. A single common password for all roles. Then update the tests to grab the values from the object.
```
env: {
      "test_pass": "password",
      "roles": {
            // info for site admin role
            "site_admin": {
                  "username": "admin_test",
                  "role": "site_admin"
            },
            // info for content editor role
            "content_editor": {
                  "username": "editor_test",
                  "role": "content_editor"
            },
            // info for content mananger role 
            "content_manager": {
                  "username": "manager_test",
                  "role": "content_manager"
            }
      }
},
```

File: testing/cypress/e2e/backend/content-editor-login.cy.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  })
+
+  after('delete user', () => {
+    cy.deleteUser(Cypress.env('content_editor'))
+  })
+
+  it('check log in', () => {
+    cy.signin(Cypress.env('content_editor'), Cypress.env('editor_password'))
+
+    cy.get('[class="title page-title"]').should('contain', 'editor_test')
```
Comment (rayestrada) [inline testing/cypress/e2e/backend/content-editor-login.cy.js:15]: Update the test to test the value from the env object.

File: testing/cypress/e2e/backend/content-manager-login.cy.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  })
+
+  after('delete user', () => {
+    cy.deleteUser(Cypress.env('content_manager'))
+  })
+
+  it('check log in', () => {
+    cy.signin(Cypress.env('content_manager'), Cypress.env('manager_password'))
+
+    cy.get('[class="title page-title"]').should('contain', 'manager_test')
```
Comment (rayestrada) [inline testing/cypress/e2e/backend/content-manager-login.cy.js:15]: Update the test to test the value from the env object.

File: testing/cypress/e2e/backend/content-manager-login.cy.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+/// <reference types="cypress" />
+
+describe('sign in as content manager', () => {
+  before('create new user', () => {
+    cy.createUser(Cypress.env('content_manager'), Cypress.env('manager_password'), Cypress.env('mnager_role'));
```
Comment (rayestrada) [inline testing/cypress/e2e/backend/content-manager-login.cy.js:5]: Typo in the last env property.

File: testing/cypress/e2e/backend/site-admin-login.cy.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  })
+
+  after('delete user', () => {
+    cy.deleteUser(Cypress.env('site_admin'))
+  })
+
+  it('check log in', () => {
+    cy.signin(Cypress.env('site_admin'), Cypress.env('admin_password'))
+
+    cy.get('[class="title page-title"]').should('contain', 'admin_test')
```
Comment (rayestrada) [inline testing/cypress/e2e/backend/site-admin-login.cy.js:15]: Update the test to test the value from the env object.

File: testing/cypress/support/commands.js [hunk 6]
```diff
@@ -1,25 +1,44 @@
...
+  );
+  //we didn’t explicitly set the failOnNonZeroExit property here and the test will fail
+  //if the given user account doesn’t exist.
+
+});
+
+
+Cypress.Commands.add('signin', (user, pass) => {
+
+  cy.visit('https://vote-gov.lndo.site/user')
```
Comment (rayestrada) [inline testing/cypress/support/commands.js:31]: update url to `http://vote-gov.lndo.site/user/login`

**Review by rayestrada — CHANGES_REQUESTED** [review]

File: testing/cypress/e2e/backend/content-manager-login.cy.js (hunk 4 above)
Comment (clmedders) [reply testing/cypress/e2e/backend/content-manager-login.cy.js:5]: fixed type

File: testing/cypress/e2e/backend/content-editor-login.cy.js (hunk 2 above)
Comment (clmedders) [reply testing/cypress/e2e/backend/content-editor-login.cy.js:15]: updated test to grab the env object

File: testing/cypress.config.js (hunk 1 above)
Comment (clmedders) [reply testing/cypress.config.js:31]: updated env format

**Comment by clmedders** [comment]
@rayestrada  this is ready for youre re-review!

**Review by rayestrada — APPROVED** [review]

**Comment by rayestrada** [comment]
@clmedders please review the documentation of Pull request standards for future PRs `/docs/standards`

**Pull request merged**